scalameta/src/main/scala/scala/meta/internal/parsers/Helpers.scala

```diff
@@ -75,11 +76,13 @@ private[meta] object Helpers {
     }
     def isCtorBody: Boolean = {
       def isSuperCall(tree: Tree): Boolean = tree match {
+        case _: Term.This => true
```

Comment:
This shouldn't work. The correct quasiquote for expr is `ctor"SomeClass(foo, bar)`.